Flow.Launcher/ViewModel/SettingWindowViewModel.cs

```diff
@@ -261,10 +267,51 @@ public IList<UserPlugin> ExternalPlugins
         {
             get
             {
-                return PluginsManifest.UserPlugins;
+                return LabelMaker(PluginsManifest.UserPlugins);
             }
         }
 
+        private  IList<UserPlugin> LabelMaker(IList<UserPlugin> list)
+        {
+            foreach (UserPlugin item in list)
+            {
+                item.LabelNew = false;
+                item.LabelInstalled = false;
+                item.LabelUpdated = false;
+
+                foreach (var vm in PluginViewModels)
+                {
+
+                    var id = vm.PluginPair.Metadata.ID;
+                    if (item.ID == vm.PluginPair.Metadata.ID) // Add Installed Label
+                    {
+                        item.LabelInstalled = true;
+                    }
+
+                    TimeSpan UpdatedDay = DateTime.Now.Subtract(item.LatestReleaseDate);
+                    int LastUpdated = UpdatedDay.Days;
+
+                    if (LastUpdated <= 5) // Add Updated Label 
+                    {
+                        item.LabelUpdated = true;
+                    }
+                }
+                TimeSpan AddedDay = DateTime.Now.Subtract(item.DateAdded);
+                int DateAdded = AddedDay.Days;
+                if (DateAdded <= 7)
+                {
+
+                    //item.LabelNew = true; // Add New Label
+                    //item.LabelUpdated = false; // Hide Updated Label when Added New Label. New and Update doesn't show both same time.
+                }
+            }
+
+            // New first, Updated second. Installed to bottom of list.
+            list = list.OrderByDescending(x => x.LabelNew).OrderByDescending(x => x.LabelUpdated).ThenBy(y => y.LabelInstalled).ToList();
```

Comment:
I believe the second one also needs to be `ThenByDescending`

Reply:
Applied